### ISSUE TITLE
[Bug]: Firefox: File download brings up 'openFile / saveToDisk' dialog box

### ISSUE THREAD

**Issue by vsravuri** [microsoft/playwright#7711]
### Playwright version

1.12.3

### Operating system

MacOS

### What browsers are you seeing the problem on?

Firefox

### Other information

Node: v14.17.0
Mac OS: 11.4
Firefox: 89

### What happened? / Describe the bug

When i try to download file in Firefox, i get a 'openFile / saveToDisk' dialog box. Same code works fine in Chrome and Webkit without prompting to open / save the file.

I followed the documentation mentioned [here](https://playwright.dev/docs/downloads) to download file.

<img width="377" alt="firefox-download" src="https://user-images.githubusercontent.com/22808594/126113203-eaac3380-ef4d-442a-9117-4492a1f2b6ac.png">
 
I can share the actual code privately if it helps you to troubleshoot the issue

### Code snippet to reproduce your bug

```shell
playwright.config.js
====================
use: {
                browserName: 'firefox',
                launchOptions: {
                    downloadsPath: path.join(__dirname, 'downloads'),
                },
                acceptDownloads: true,
            }

example.spec.js
===============
let elmFrame = page.frames().find(frame => frame.name() === 'submainframe');

        const [download] = await Promise.all([
            page.waitForEvent('download'), // wait for download to start
            elmFrame.click('css=tr > td >> text=PlaywrightTest >> xpath=parent::td/parent::tr >> td >> css=img[alt="Download"]'),
        ]);

        let downloadPath;
        downloadPath = await download.path();
        console.log(downloadPath);
```

### Relevant log output

```shell
n/a
```

**Comment by aslushnikov** [comment]
@vsravuri could you please share a repro that we can run locally to reproduce this?

**Comment by vsravuri** [comment]
@aslushnikov Sure, i will send you the details shortly.

**Comment by aslushnikov** [comment]
> Sure, i will send you the details shortly.

@vsravuri  I unfortunately didn't get the archive - instead, I received "Your organization's email policy doesn't permit this type of file. If you need it, please contact your administrator.". 

Maybe you can create a private repo and give access to it?

**Comment by vsravuri** [comment]
@aslushnikov I have sent you the code

**Comment by vsravuri** [comment]
@aslushnikov Please let me know if you need additional information.

**Comment by aslushnikov** [comment]
@vsravuri yep, I received your files. I'll take a look!

**Comment by vsravuri** [comment]
@aslushnikov Have you got a chance to look at it ?

**Comment by vsravuri** [comment]
@aslushnikov Appreciate if this issue is resolved in v1.15

**Comment by mxschmitt** [comment]
@vsravuri the fix landed on tip-of-tree and should be available in our canary right now: `npm i @playwright/test@next`

And will also be published in the future v1.15

Would be awesome if you could verify.

**Comment by aslushnikov** [comment]
Should be fixed with #8016; please re-file if it isn't!
